ammonite-repl (2.13-240) Reversion to stable

TJC opened the pull request:
This commit reverts Ammonite-REPL from the Scala-3-based version, to
the Scala-2.13 version. It's still the same version of Ammonite, just
compiled against a different version of Scala.

This is because under Scala 3, one of Ammonite's core features is currently broken.
This commit also improves the formula test, to demonstrate the issue that fails against version 3.0-2.40 of ammonite-repl.

See also:
- https://github.com/Homebrew/discussions/discussions/2117
- https://github.com/com-lihaoyi/Ammonite/issues/1190

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

chenrui333 reviewed (commented):
LGTM
cc @steinybot

carlocab commented on Formula/ammonite-repl.rb:
```diff
@@ -1,10 +1,11 @@
   desc "Ammonite is a cleanroom re-implementation of the Scala REPL"
   homepage "https://ammonite.io/"
-  url "https://github.com/lihaoyi/Ammonite/releases/download/2.4.0/3.0-2.4.0"
+  url "https://github.com/lihaoyi/Ammonite/releases/download/2.4.0/2.13-2.4.0"
```
Can we add a comment here that explains which version of Ammonite to use? i.e. between the `3.0-x` or `2.13-x`, etc versions.

TJC replied:
I've added some notes now

carlocab commented on Formula/ammonite-repl.rb:
```diff
@@ -19,7 +20,8 @@ def install
   test do
-    output = shell_output("#{bin}/amm -c 'print(\"hello world!\")'")
-    assert_equal "hello world!", output.lines.last
+    (testpath/"testscript.sc").write("#!/usr/bin/env amm\n@main\ndef fn(): Unit = {\nprintln(\"hello world!\")\n}\n")
```
```suggestion
    (testpath/"testscript.sc").write <<~EOS
      #!/usr/bin/env amm
      @main
      def fn(): Unit = {
      println("hello world!")
      }
    EOS
```
Why do we need to change this?

TJC replied:
I changed the test to one that is slightly more comprehensive. This test fails on the "bad" version of Ammonite, and passes on the "good" version.
Once the latest versions are fixed, presumably this test will pass again, and allow us to upgrade it here.

carlocab commented on Formula/ammonite-repl.rb:
```diff
@@ -19,7 +20,8 @@ def install
-    assert_equal "hello world!", output.lines.last
+    (testpath/"testscript.sc").write("#!/usr/bin/env amm\n@main\ndef fn(): Unit = {\nprintln(\"hello world!\")\n}\n")
+    output = shell_output("#{bin}/amm #{testpath}/testscript.sc")
+    assert_match "hello world!", output.lines.last
```
```suggestion
    assert_equal "hello world!", output.lines.last.chomp
```

TJC commented:
I've cleaned up the file and added some more explanatory notes.

There are several commits in the PR now. Let me know if you would like me to squash them back to a single commit?

carlocab commented:
> Let me know if you would like me to squash them back to a single commit?

Go ahead. Use the commit message
```
ammonite-repl: revert to stable version
```
or similar, as long as it starts with `ammonite-repl:`.

TJC commented:
> > Let me know if you would like me to squash them back to a single commit?
> 
> Go ahead. 

Done.

carlocab reviewed (approved):
Thanks, @TJC.
